### PR TITLE
Update UG for v1.4

### LINKED ISSUE #168: [PE-D] Add m command does not give unkown command error
How to reproduce: For some unkown command, it will still work. FOr example: add a n/Test m/1; 
Expected: Give an error message of unknown command;
Actual: proceed to run the command, treated it as add assessment. 

-------------
Labels: `type.FunctionalityBug` `severity.VeryLow`
original: zunedz/ped#2

### LINKED ISSUE #169: [PE-D] Email format does not account for certain valid emails
Regular expressions for emails are notoriously hard, so I won't blame you for this, but there are some (obviously valid) email addresses that are rejected by TAssist when adding students. Email addresses with punycoded domains are one such example.

Domains can have *some* non-ASCII characters within them, and the most obvious examples are [IDN ccTLDs](https://en.wikipedia.org/wiki/Internationalized_country_code_top-level_domain), such as:  
- .укр
- .бг
- .中国

Take [網路家庭.台灣](http://網路家庭.台灣) for example. That's the domain for a Taiwanese web portal (?) named PChome, and the punycoded variant is [xn--fctw2evy5bdno.xn--kpry57d](http://xn--fctw2evy5bdno.xn--kpry57d).

An email address with these domains are rejected because they contain more than one consecutive hyphens, even though they are ultimately valid email addresses.

(Note that this means that *any* valid domain names with two hyphens, e.g. `z--z.com`, are rejected too.)

This is ultimately not too much of an issue since most people don't have email addresses with these kinds of domains, hence the VeryLow severity, but it's something you should consider.

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: zhongfu/ped#1

### LINKED ISSUE #171: [PE-D] Add s command does not give unkown command error
How to reproduce: use Add s command
expected: give an error message
actual: the program treated it as add student 

-------------
Labels: `type.FunctionalityBug` `severity.VeryLow`
original: zunedz/ped#3

### LINKED ISSUE #173: [PE-D] Add C command does not give unkown command error
how to reproduce: input `add c`
expected: Unknown command or Unknown entity;
Actual: The command is treated as add class command;

-------------
Labels: `type.FunctionalityBug` `severity.VeryLow`
original: zunedz/ped#4

### LINKED ISSUE #177: [PE-D] Inconsistent examples between UG and error message for list assessment in Invalid command format!
-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: sibinhho99-nus/ped#2

### LINKED ISSUE #187: [PE-D] Some commands with invalid formats are used in the UG, under the section describing the  `delete` commands
e.g.:

- `list student c\1`
- `list class \m 2`

Marked as low sev because:
- the commands technically still run (although without the intended effect, i.e. that the list is filtered)
- the rest of the sentence ("... deletes the 2nd class group in the **filtered** class group list (filtered by module index 2) in TAssist.") still holds, as the 2nd class group in the resulting list will still be deleted. (Though the class group list is technically not filtered.)

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: zhongfu/ped#11

### LINKED ISSUE #191: [PE-D] [Suggestion] Grade Values and Representation
## Suggestion
Shouldn't there be a note to the reader what are the grade values and what they represent.

## Screenshots

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: Ardentsoul/ped#6

### LINKED ISSUE #204: [PE-D] Unable to edit the saved data
How to reproduce: Use the application. For context, I'm using java 11 on M1 MacBook.
Expected: I could find the data on the folders the jar is in
Actual: I could not find the data

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: zunedz/ped#12

### LINKED ISSUE #206: [PE-D] (UG) Students could be plural in grading assessments
Multiple students can be graded in a single command, so students could perhaps be plural

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: sibinhho99-nus/ped#11

### LINKED ISSUE #208: [PE-D] Incorrect constraint in Userguide for delete assessment command
At bullet point 2, it should be `the index refers to the index number shown in the displayed assessment list`

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: zunedz/ped#14

### LINKED ISSUE #210: [PE-D] Buttons in the application is not mentioned in the user guide
Buttons such as Check absentee and See attempts are not mentioned/explained in the user guide. Overall, the team has done a very good job and there is no major bug I found on the app. 

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: zunedz/ped#16

### PR DESCRIPTION
**Pull request by Gernene**
closes #168 
closes #171 
closes #173 
closes #169 
closes #177 
closes #187 
closes #191 
closes #204 
closes #206 
closes #208 
closes #210

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -3,13 +3,39 @@ layout: page
...
+1. Type the command in the command box and press Enter to execute it. e.g. typing **`list student`** and pressing Enter will list all students added.<br>
+   Some example commands you can try:
+
+   * **`list student`**: Lists all students.
+
+   * **`add student`** `id/E0123456 n/John Doe e/johnd@u.nus.edu`: Adds a student named `John Doe` to TAssist.
+
+   * **`delete student`** `3`: Deletes the 3rd student shown in the listing of the entity.
+
+1. Refer to the 5rr below for details of each command.
```
Comment (jxt00) [inline docs/UserGuide.md:35]: 5rr?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -39,6 +65,23 @@ TAssist is a **desktop app for managing students and their participation in less
...
+
+![Ui](images/Ui.png)
+
+From the class group list, click the "Check Attendance" button to open a popup window. This popup  displays a list of lessons 1-13 and any students who were **absent** for a lesson.
+
+![Ui](images/Ui_Check_Attendance.jpg)
+
+From the assessment list, click the "See Attempts" button to open a popup window. This popup displays a list of student attempts for that assessment including their name and assigned grade.
+
+![Ui](images/Ui_See_Attempts.jpg)
```
Comment (jxt00) [inline docs/UserGuide.md:83]: Might be better if the pictures are smaller to match the size of the UI shown in `Ui.png`.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -228,147 +243,146 @@ Format: `add module n/MODULE_NAME c/MODULE_CODE a/ACADEMIC_YEAR`
...
-### Taking student attendance
 
-#### Marking attendance: `mark`
+### Managing Classes
 
-Marks student(s)' attendance(s).
+#### Add a class: `add class`
 
-Format: `mark c/CLASS_GROUP_INDEX w/WEEK_INDEX s/all|STUDENT_INDEXES|STUDENT_IDS`
+Add class groups to modules you've created. Class groups refer to tutorials, labs, and sectionals you personally teach.
```
Comment (jxt00) [inline docs/UserGuide.md:317]: Recitations as well.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -380,155 +394,167 @@ Examples:
...
 
 
-##### Filtering class groups
+#### Delete a student: `delete student`
 
-Shows a list of class groups belonging to a module.
+Delete the specified student from TAssist as well as the student's attempt(s) in the assessment(s).
 
-Format: `list class m/MODULE_INDEX`
+Deletes the specified module from TAssist as well as its associated class group(s) and assessment(s).
```
Comment (jxt00) [inline docs/UserGuide.md:419]: Extra line.

File: docs/UserGuide.md (hunk 1 above)
Comment (Gernene) [reply docs/UserGuide.md:35]: I have no idea what happened here. Fixing right now.

**Comment by jxt00** [comment]
Line 108: Description for the `KEYWORD` parameter is wrong.

**Review by lkldev — APPROVED** [review]
LGTM

**Review by jxt00 — APPROVED** [review]
LGTM

**Comment by lkldev** [comment]
closes #168, closes #171, closes #173, closes #169, closes #177, closes #187, closes #191, closes #204, closes #206, closes #208, closes #210,

**Pull request merged**